zerver/views/home.py

```diff
@@ -254,6 +255,23 @@ def home_real(request: HttpRequest) -> HttpResponse:
                                'embedded': narrow_stream is not None,
                                },)
     patch_cache_control(response, no_cache=True, no_store=True, must_revalidate=True)
+
+    uri_host_source = urlparse(user_profile.realm.uri).netloc
+    ws_source = ''
+    if uri_host_source:
+        ws_source = ('ws://%s/' % (uri_host_source))
+    csp_policy = ("object-src 'none'; "
+                  "default-src 'none'; "
+                  "img-src https://* 'self'; "
```

Comment:
This one may want to use `EXTERNAL_URI_SCHEME` or something so that `http` works in development?

Reply:
the https in image src is for external links pasted which might trigger image previews.

Any chance of a side effect that the intention here?

Reply:
I guess I get might be troubling here, adding 'self' to the whitelist doesn't really make sense, since https:// is kinda bigger scope than 'self'